### PR TITLE
Feature/init cosys integration

### PR DESCRIPTION
**Pull request by simonhir**
**Description**

Init cosys integration.

**Reference**

Issues #122

### CODE REVIEW

File: refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-example/src/main/resources/application-local.yml
Comment (simonhir) [inline refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-example/src/main/resources/application-local.yml:1]: Not really nice, but isn't easy to test, so would let it this way and refactor this within #133

File: refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-core/pom.xml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+        <!-- Other -->
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-lang3</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.google.code.gson</groupId>
+            <artifactId>gson</artifactId>
+            <version>2.11.0</version>
```
Comment (DanielOber) [inline refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-core/pom.xml:33]: can you move all version numbers to the props

Comment (simonhir) [reply refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-core/pom.xml]: Done

File: refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-core/src/test/java/de/muenchen/refarch/integration/cosys/adapter/out/s3/S3AdapterTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        s3Adapter.saveDocumentInStorage("test/filePath.txt", DATA_AS_BYTE_ARRAY);
+        verify(documentStorageFileRepository).saveFile("test/filePath.txt", DATA_AS_BYTE_ARRAY, 1);
+        verifyNoMoreInteractions(s3FileTransferRepository);
+
+    }
+
+    @Test
+    void saveDocumentInStorageWithThrowsDocumentStorageException()
+            throws DocumentStorageException, DocumentStorageClientErrorException, DocumentStorageServerErrorException {
+        // v2
```
[the remark this answers is not in the document]
Comment (simonhir) [reply refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-core/src/test/java/de/muenchen/refarch/integration/cosys/adapter/out/s3/S3AdapterTest.java]: Done.
Forgot to remove it on cleanup.

File: refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-starter/src/main/resources/application-cosys-client.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+            user-info-uri: ${SSO_COSYS_ISSUER_URL}/protocol/openid-connect/userinfo
+            jwk-set-uri: ${SSO_COSYS_ISSUER_URL}/protocol/openid-connect/certs
+            user-name-attribute: user_name
+        registration:
+          cosys:
+            provider: cosys
+            authorization-grant-type: client_credentials
+            client-id: ${SSO_COSYS_CLIENT_ID}
+            client-secret: ${SSO_COSYS_CLIENT_SECRET}
+            scope: email, profile, openid  # needed for userInfo endpoint
```
Comment (DanielOber) [inline refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-starter/src/main/resources/application-cosys-client.yml:17]: would it be better to move that to the documentation, like you did it for the s3?

Comment (simonhir) [reply refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-starter/src/main/resources/application-cosys-client.yml:17]: Ah yeah good idea. Was about to look for this 👍

Comment (simonhir) [reply refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-starter/src/main/resources/application-cosys-client.yml]: Done

**Review by DanielOber — DISMISSED** [review]
LGTM

**Comment by simonhir** [comment]
Waiting for #208

File: refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-core/src/test/java/de/muenchen/refarch/integration/cosys/application/usecase/CreateDocumentUseCaseTest.java
Comment (devtobi) [inline refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-core/src/test/java/de/muenchen/refarch/integration/cosys/application/usecase/CreateDocumentUseCaseTest.java:1]: You should use `@ExtendWith(MockitoExtension.class)` and `@InjectMocks` to be aligned with the other test classes.

File: refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-example/src/main/java/de/muenchen/refarch/integration/cosys/example/NoSecurityConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import org.springframework.security.config.annotation.web.configurers.HeadersConfigurer;
+import org.springframework.security.web.SecurityFilterChain;
+
+@Configuration
+@EnableWebSecurity
+public class NoSecurityConfiguration {
+
+    @Bean
+    public SecurityFilterChain mainSecurityFilterChain(final HttpSecurity http) throws Exception {
+        // @formatter:off
```
Comment (devtobi) [inline refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-example/src/main/java/de/muenchen/refarch/integration/cosys/example/NoSecurityConfiguration.java:17]: Why is formatter suppression needed here?

File: refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-example/src/main/resources/application-local.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
+refarch:
+  s3:
+    client:
+      document-storage-url: http://localhost:8086
+      enable-security: true
+  cosys:
+    url: https://cosysk007.srv.muenchen.de/cosys-eai/rest
```
Comment (devtobi) [inline refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-example/src/main/resources/application-local.yml:7]: Can we somehow get rid of that internal reference?

File: refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-starter/src/main/java/de/muenchen/refarch/integration/cosys/configuration/CosysProperties.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    @Setter
+    public static class MergeProperties {
+
+        private String datafile;
+
+        private String inputLanguage;
+
+        private String outputLanguage;
+
+        private String keepFields;
```
Comment (devtobi) [inline refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-starter/src/main/java/de/muenchen/refarch/integration/cosys/configuration/CosysProperties.java:32]: Do we need validations on those properties as well?

**Review by devtobi — CHANGES_REQUESTED** [review]

File: refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-core/src/test/java/de/muenchen/refarch/integration/cosys/adapter/out/s3/S3AdapterTest.java
[the remark this answers is not in the document]
Comment (simonhir) [reply refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-core/src/test/java/de/muenchen/refarch/integration/cosys/adapter/out/s3/S3AdapterTest.java:1]: I like this war more as then the variables can be final.
Aligned others to this syntax

File: refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-starter/src/main/java/de/muenchen/refarch/integration/cosys/configuration/CosysProperties.java (hunk 6 above)
Comment (simonhir) [reply refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-starter/src/main/java/de/muenchen/refarch/integration/cosys/configuration/CosysProperties.java:32]: Done

File: refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-client/src/main/api/api-docs.json [hunk 7]
```diff
@@ -0,0 +1,1838 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "description": "This application provides the routing for calls from Fachanwendungen to coSys.",
```
[the remark this answers is not in the document]
Comment (simonhir) [reply refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-client/src/main/api/api-docs.json:4]: I understand your mentions of the open API spec of internal urls and typos but in my opinion we shouldn't change a externally provided spec.

File: refarch-integrations/refarch-cosys-integration/README.md
[the remark this answers is not in the document]
Comment (simonhir) [reply refarch-integrations/refarch-cosys-integration/README.md:1]: Added the link

File: refarch-integrations/README.md
[the remark this answers is not in the document]
Comment (simonhir) [reply refarch-integrations/README.md:1]: I added the link to the inner readme think thats enough and more distracting here

File: refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-client/src/main/api/api-docs.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (devtobi) [reply refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-client/src/main/api/api-docs.json:4]: Ok that's fine. :)

File: refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-example/src/main/resources/application-local.yml (hunk 5 above)
Comment (simonhir) [reply refarch-integrations/refarch-cosys-integration/refarch-cosys-integration-example/src/main/resources/application-local.yml]: Done

**Review by devtobi — APPROVED** [review]
LGTM

**Pull request merged**